### ISSUE TITLE
Wrong read/write configuration of memory object

### ISSUE THREAD

**Issue by L30nardoSV** [ccsb-scripps/AutoDock-GPU#32]
This error was pointed by [oclgrind](https://github.com/jrprice/Oclgrind/wiki/Using-Oclgrind) in `kernel4`:

```zsh
Invalid write to read-only buffer
	Kernel: gpu_gen_and_eval_newpops
	Entity: Group(47,0,0)
	  call spir_func void @_Z17wait_group_eventsiP9ocl_event(i32 1, %opencl.event_t** nonnull %ev171) #9, !dbg !449
	At line 2097 (column 3) of input.cl:
	  wait_group_events(1,&ev);
```

So `mem_dockpars_conformations_current` must be configured as `CL_MEM_READ_WRITE`, 
and **NOT** just for reading as in `master` and in `debugfastergrad`.

Although the source code in `kernel4` never updates this memory object, it actually updates it in every other genetic iteration because populations are updated by switching pointers as in `/host/src/performdocking.cpp`.

**Comment by L30nardoSV** [comment]
For master branch, this is corrected in commit 25187c8353cdd189ca32051937e3a11e9a1c8236.

**Comment by L30nardoSV** [comment]
For debugfasterbranch, this is corrected in commit e4374d852467b8babf9f0b9011221565c9937a92.
